services/qan/client.go

```diff
@@ -125,19 +127,52 @@ func (c *Client) Collect(ctx context.Context, req *qanpb.CollectRequest) error {
 			c.l.Error(err)
 			continue
 		}
+		mb := &qanpb.MetricsBucket{
+			Queryid:              m.Common.Queryid,
+			Fingerprint:          m.Common.Fingerprint,
+			ServiceName:          service.ServiceName,
+			Database:             m.Common.Database,
+			Schema:               m.Common.Schema,
+			Tables:               m.Common.Tables,
+			Username:             m.Common.Username,
+			ClientHost:           m.Common.ClientHost,
+			ServiceType:          string(service.ServiceType),
+			AgentId:              m.Common.AgentId,
+			AgentType:            m.Common.AgentType,
+			PeriodStartUnixSecs:  m.Common.PeriodStartUnixSecs,
+			PeriodLengthSecs:     m.Common.PeriodLengthSecs,
+			Example:              m.Common.Example,
+			ExampleFormat:        qanpb.ExampleFormat(m.Common.ExampleFormat),
+			ExampleType:          qanpb.ExampleType(m.Common.ExampleType),
```

Comment:
That code will break if one enum will change a value, but another will not. We need an honest switch for both.